.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: CI Check
+on: [push, pull_request, workflow_dispatch]
+jobs:
+  clean-and-test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ['3.10', '3.11']
+    steps:
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Getting repository
+      uses: actions/checkout@v4
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```

Comment:
You generally don't need this, if you pip install the package below, that should give you all the requirements.

Reply:
So just leave out the 'pip install -r ...' ?

Reply:
Only the `pip install -r requirements.txt`. Instead, you can directly put `pip install -e .`